Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -42,8 +42,9 @@ def cmd_output(*args)
     cmd_args << "-rintegration_mocks"
     cmd_args << (HOMEBREW_LIBRARY_PATH/"../brew.rb").resolved_path.to_s
     cmd_args += args
+    cmd_exec = [RUBY_PATH] + cmd_args
     Bundler.with_original_env do
-      ENV["HOMEBREW_BREW_FILE"] = HOMEBREW_PREFIX/"bin/brew"
+      ENV["HOMEBREW_BREW_FILE"] = cmd_exec.join " "
```

Comment:
One problem is that these instances re-use the same test identifier and thus overwrite each other's coverage data when running with `--coverage`. There might be more issues, but to be honest, I haven't spent much time thinking about this.